### ISSUE TITLE
status.data.humancellatlas.org

### ISSUE THREAD

**Issue by rhiananthony** [HumanCellAtlas/dcp#83]
Create a website which indicates the current status of the DCP, hosted outside of the regular DCP infrastructure.

**Comment by rhiananthony** [comment]
Kylee - June 5, 2018 - 11:17AM
Can we use the grafana dashboard Matt already made to fill in a status page that faces externally. Or we need something to read-only check the health endpoints and fill in a page.
Could be done manually.
Can we get some design help on this?


Kylee - May 8, 2018 - 11:23AM
Can we use Grafana as an API to figure out if the DCP is up or down period, and then make a small dynamic status page. Maybe with an iframe that would expose that without credentials?

**Comment by mweiden** [comment]
There's a (M)MVP version of this page here https://humancellatlas.github.io/

This needs improvement, there's an AWS rate limit that I need to design around.

**Comment by mweiden** [comment]
I've fixed the scalability issues here. This is now looking for a home and design support.

@kbergin do you have ideas on how we can move forward on this?

**Comment by kbergin** [comment]
Let's see - first off, this is incredibly useful and I'm so happy it exists!

Things I could think of as possible iterations/open questions that influence its future: 
- Add prod to the list for each service
- Is this info already available (for prod) to the data portal? or at least for the DCP level up time, can we automate a message banner being posted on the website when there is downtime?
- Is this website intended to be public facing or for operations? 
- Does it need better design or does it satisfy the need as is (if it's for internal use, the answer seems simpler than if it's user facing)
- Would this ideally exist as a page on the data portal? 
- Is there any way to group together and have one dashboard for the operations of DCP? Maybe it's not confusing to other people, but I'm starting to have trouble keeping track between grafana, kibana, allspark, gitlab, this page.. while they are awesome, they also add up. Or if this page could have links out to each of those other services and act as a home base? 

It seems like ownership of this would fall into a similar realm as either 1) the data portal if it's user facing or 2) the logging/grafana/gitlab/integration test/alerting type services. Is the latter intended to be taken over by a particular group or is it a shared ownership across teams? 

Open to a lot of things here, but as far as first iteration that meets the use case, the immediate benefit of seeing the overall status is clearly met here. I think when this feature first came up, the idea was to have a public facing 'is prod up yes/no' status page, which I don't think this satisfies. Unclear to me if the intention was to add prod to this page or have a separate page for that use case. 

Overall thoughts?

**Comment by mweiden** [comment]
@kbergin 

tl;dr - I think we should have a polished, separate page for prod, maybe incorporated in data portal for the public.

Here is more in excruciating detail, point by point.
* Extending to prod: definitely, I'll work on that. Regarding your internal/external questions this should probably be separate as end-users will not care about earlier deployment environments.
* Data portal: as soon as they have a GitLab build and a health check endpoint, sure
* Banner: would probably be a separate feature
* Public/internal: this is meant to be public. In fact, he real value-add here is the [status api](https://github.com/HumanCellAtlas/status-api) which makes the internal health and availability info public. We can access the status badges from it and put them anywhere.
* Design: I think it can be for internal use during the pilot phase, but when the project is opened to the public it should be more polished like [others](https://www.google.com/appsstatus#hl=en&v=status) [out](https://status.aws.amazon.com/) [there](https://azure.microsoft.com/en-us/status/).
* This could exist on the data portal domain, but should be hosted elsewhere in case the main data portal goes down.
* Re. the multiplicity of services. This really serves a different purpose than the others.
    * GitLab - CI/CD
    * Grafana - internal dashboards
    * Kibana - just use it to search logs, forget the dashboards there
    * Status API - badges for public status
    * humancellatlas.github.io - this is really just a sample of how you can use the Status API... really the same info can go anywhere
* Re. a centralized dashboard.
    * For internal purposes, we have the account-wide dev [[1](https://metrics.dev.data.humancellatlas.org/d/LrYJeR2iz/dev-account-status?refresh=1m&orgId=1), [2](https://metrics.dev.data.humancellatlas.org/d/account-dev/account-status-hca?refresh=1m&orgId=1)] (pick your flavor) and [prod](https://metrics.data.humancellatlas.org/d/v4-0_FWiz/dcp-health-prod?refresh=1m&orgId=1) dashboards on Grafana.
    * For external purposes, currently we have humancellatlas.github.io... I guess it's up to PM how we want to package the public info
* Ownership: I'd say DevSecOps/Arch ultimately owns this as it's cross-cutting DevOps tooling

**Comment by kbergin** [comment]
@mweiden Thanks so much for explaining! 

Good point about putting the page on the data portal, if it's down that doesn't help! I'll make an epic of things surrounding this topic and bring it up at the PM meeting at 11. I'd agree that the general priority is probably a bit lower until we're in a full launch (i.e. maybe we can get by without it being user facing until the CBeta is over?)

I'll tag you on that epic for review.

**Comment by mweiden** [comment]
@kbergin Yes, I believe we can get by without it being user facing in CBeta. I will put this ticket back in "ToDo." I might make a couple small adjustments to the site I have up while we're waiting.

**Comment by mweiden** [comment]
@kbergin Heads up: I added the prod deployment, so we can now retrieve the prod build and service checks now as well.

I've also added this to the draft page.

**Comment by kbergin** [comment]
@mweiden thanks! Given the epic linked that has the follow up work from this ticket, I'd be inclined to call this ticket complete. Thoughts?

**Comment by mweiden** [comment]
@kbergin I agree. Closing this now.
